File: src/FactoryMethod/src/Episodio.java
public class Episodio {
    private String titulo;
    private int numero;
    private int duracao;
    private String descricao;

    public Episodio(String titulo, int numero, int duracao, String descricao) {
        this.titulo = titulo;
        this.numero = numero;
        this.duracao = duracao;
        this.descricao = descricao;
    }

    // Métodos getter e setter
    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public int getDuracao() {
        return duracao;
    }

    public void setDuracao(int duracao) {
        this.duracao = duracao;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    // Método para imprimir informações do episódio
    public void printarInformacoes() {
        System.out.println("Episódio " + numero + ": " + titulo + " (" + duracao + " minutos)");
        System.out.println("Descrição: " + descricao);
    }
}

File: src/FactoryMethod/src/models/episodio.ts
export class Episodio {
    private titulo: string;
    private numero: number;
    private duracao: number;
    private descricao: string;

    constructor(titulo: string, numero: number, duracao: number, descricao: string) {
        this.titulo = titulo;
        this.numero = numero;
        this.duracao = duracao;
        this.descricao = descricao;
    }

    // Métodos getter e setter
    public getTitulo(): string {
        return this.titulo;
    }

    public setTitulo(titulo: string): void {
        this.titulo = titulo;
    }

    public getNumero(): number {
        return this.numero;
    }

    public setNumero(numero: number): void {
        this.numero = numero;
    }

    public getDuracao(): number {
        return this.duracao;
    }

    public setDuracao(duracao: number): void {
        this.duracao = duracao;
    }

    public getDescricao(): string {
        return this.descricao;
    }

    public setDescricao(descricao: string): void {
        this.descricao = descricao;
    }

    // Método para imprimir informações do episódio
    public printarInformacoes(): void {
        console.log(`Episódio ${this.numero}: ${this.titulo} (${this.duracao} minutos)`);
        console.log(`Descrição: ${this.descricao}`);
    }
}